skema/skema-rs/mathml/src/expression.rs

```diff
@@ -14,42 +23,1094 @@ enum Atom {
 #[derive(Debug, PartialEq, Clone)]
 enum Expr {
     Atom(Atom),
-    Expression { op: Operator, args: Vec<Expr> },
+    Expression {
+        op: Vec<Operator>,
+        args: Vec<Expr>,
+        name: String,
+    },
+}
+
+#[derive(Debug, PartialEq, Clone)]
```

Comment:
Can you add a short docstring for the `PreExp` struct explaining what it is?

Reply:
Sure! This is an Intermediate data structure to support the generation of graphs of mathematical expressions.

Reply:
Can you add the explanation as a doc comment above the struct?

Reply:
Sorry, just saw that it is there, nevermind!